contracts/FlowTransactionSchedulerUtils.cdc

```diff
@@ -318,10 +319,21 @@ access(all) contract FlowTransactionSchedulerUtils {
             let pastTimestamps = self.sortedTimestamps.getBefore(current: currentTimestamp)
             for timestamp in pastTimestamps {
                 let ids = self.idsByTimestamp[timestamp] ?? []
+                if ids.length == 0 {
+                    self.sortedTimestamps.remove(timestamp: timestamp)
+                    continue
+                }
                 for id in ids {
                     let status = FlowTransactionScheduler.getStatus(id: id)
-                    if status == nil || status == FlowTransactionScheduler.Status.Unknown {
+                    if status == nil || status! != FlowTransactionScheduler.Status.Scheduled {
                         transactionsToRemove.append(id)
+                        // Need to temporarily limit the number of transactions to remove
+                        // because some managers on mainnet have already hit the limit and we need to batch them
+                        // to make sure they get cleaned up properly
+                        // This will be removed eventually
+                        if transactionsToRemove.length > 50 {
```

Comment:
This limit is only temporary. The cases that we've become aware of going over the computation limit were in the 100s of iterations, so we believe that this should be sufficient to allow them to clean up without hitting the limit again. We'll be doing some more testing also to make sure